lib/utilities/pristine.js

```diff
@@ -142,11 +148,19 @@ function nodeModulesSetup(options) {
     promise = promise
       .then(() => {
         debug('installing ember-disable-prototype-extensions');
-        return runCommand('npm', 'install', 'ember-disable-prototype-extensions');
+        if (options.yarn) {
+          return runCommand('yarn', 'add', 'ember-disable-prototype-extensions');
+        } else {
+          return runCommand('npm', 'install', 'ember-disable-prototype-extensions');
+        }
       })
       .then(() => {
         debug("installed ember-disable-prototype-extension");
-        return runCommand('npm', 'install');
+        if (options.yarn) {
+          return runCommand('yarn');
+        } else {
+          return runCommand('npm', 'install');
+        }
```

Comment:
Following up on my comment https://github.com/tomdale/ember-cli-addon-tests/pull/105/files#r118477645 and I think also what @kellyselden was suggesting, if this addon knows which package manager to use, maybe we can abstract this and the `yarn add` thing above away into a reusable method and make it public (like `app.install()` and `app.install('my-addon') `) which users of this addon can use to customize their test setup without having to care about which package manager shall be used (e.g. here https://github.com/kaliber5/ember-fastboot-addon-tests/blob/master/lib/util/app-manager.js#L41)? 

Can be a separate PR as well, just wanted to mention my thoughts here...